src/main/java/org/motechproject/prevac/service/impl/VisitScheduleServiceImpl.java

```diff
@@ -44,14 +47,14 @@
                 primeVacDate = subject.getPrimerVaccinationDate();
             }
 
-            LocalDate screeningDate = getScreeningDate(subject);
-            if (screeningDate != null) {
+            Map<String, LocalDate> visitDates = getScreeningAndPrimeVacDates(subject);
+            if (!visitDates.isEmpty()) {
                 if (subject.getFemaleChildBearingAge() == null || !subject.getFemaleChildBearingAge()) {
```

Comment:
I think you should add one more if (plannedDate != null) and set earliestDate = plannedDate there, ale else if the rest without change